lib/usdUfe/ufe/trf/UsdPointInstanceOrientationModifier.cpp

```diff
@@ -16,73 +16,57 @@
 } // namespace
 
-namespace MAYAUSD_NS_DEF {
-namespace ufe {
+namespace USDUFE_NS_DEF {
 
 // Ensure that UsdPointInstanceOrientationModifier is properly setup.
 using base = UsdPointInstanceModifierBase<Ufe::Vector3d, PXR_NS::GfQuath>;
-MAYAUSD_VERIFY_CLASS_BASE(base, UsdPointInstanceOrientationModifier);
-MAYAUSD_VERIFY_CLASS_VIRTUAL_DESTRUCTOR(UsdPointInstanceOrientationModifier);
-MAYAUSD_VERIFY_CLASS_NOT_MOVE_OR_COPY(UsdPointInstanceOrientationModifier);
+USDUFE_VERIFY_CLASS_BASE(base, UsdPointInstanceOrientationModifier);
+USDUFE_VERIFY_CLASS_VIRTUAL_DESTRUCTOR(UsdPointInstanceOrientationModifier);
+USDUFE_VERIFY_CLASS_NOT_MOVE_OR_COPY(UsdPointInstanceOrientationModifier);
 
 /* override */
 GfQuath UsdPointInstanceOrientationModifier::convertValueToUsd(const Ufe::Vector3d& ufeValue) const
 {
-    const MEulerRotation eulerRotation(
-        MAngle(ufeValue.x(), MAngle::kDegrees).asRadians(),
-        MAngle(ufeValue.y(), MAngle::kDegrees).asRadians(),
-        MAngle(ufeValue.z(), MAngle::kDegrees).asRadians());
-
-    MTransformationMatrix transformMatrix(MTransformationMatrix::identity);
-    transformMatrix.rotateTo(eulerRotation);
-
-    double real;
-    double i, j, k;
-    transformMatrix.getRotationQuaternion(i, j, k, real);
-
-    return GfQuath(
-        static_cast<float>(real),
-        static_cast<float>(i),
-        static_cast<float>(j),
-        static_cast<float>(k));
+    // Input vector from Ufe is in degrees.
+    PXR_NS::GfVec3f eulerXYZ(ufeValue.x(), ufeValue.y(), ufeValue.z());
+
+    // Create a rotation (XYZ) from the Euler angles.
+    auto rotD = PXR_NS::GfRotation(GfVec3d::XAxis(), eulerXYZ[0])
+        * PXR_NS::GfRotation(GfVec3d::YAxis(), eulerXYZ[1])
+        * PXR_NS::GfRotation(GfVec3d::ZAxis(), eulerXYZ[2]);
+
+    auto            usdQuaternionD = rotD.GetQuaternion();
+    PXR_NS::GfQuath usdQuatD(rotD.GetQuat());
+    return usdQuatD;
 }
 
 /* override */
 Ufe::Vector3d UsdPointInstanceOrientationModifier::convertValueToUfe(const GfQuath& usdValue) const
 {
-    MTransformationMatrix transformMatrix(MTransformationMatrix::identity);
-
-    const GfVec3h& imaginary = usdValue.GetImaginary();
-    transformMatrix.setRotationQuaternion(
-        imaginary[0u], imaginary[1u], imaginary[2u], usdValue.GetReal());
-
-    const MEulerRotation eulerRotation = transformMatrix.eulerRotation();
+    // First create a rotation from the quaternion.
+    PXR_NS::GfRotation gfRot(usdValue);
 
-    return Ufe::Vector3d(
-        MAngle(eulerRotation.x, MAngle::kRadians).asDegrees(),
-        MAngle(eulerRotation.y, MAngle::kRadians).asDegrees(),
-        MAngle(eulerRotation.z, MAngle::kRadians).asDegrees());
+    // Convert the rotation to Euler XYZ (when we decompose we use ZYX).
+    auto          eulerXYZ = gfRot.Decompose(GfVec3d::ZAxis(), GfVec3d::YAxis(), GfVec3d::XAxis());
+    Ufe::Vector3d newRet(eulerXYZ[2], eulerXYZ[1], eulerXYZ[0]);
+    return newRet;
```

Comment:
Use USD API to covert value (instead of Maya API). I verified (using asserts from the tests) that the computed result was the same.

Reply:
I don't understand why we decompose ZYX and create a XYZ. (Also, i'd have named the variable eulerZYX since that is its contents.)

Reply:
I also don't understand this, but I found that from existing code in our repo.
https://github.com/Autodesk/maya-usd/blob/dev/lib/mayaUsd/fileio/translators/translatorXformableDecompose.cpp#L45

It says, "Assumes rotationOrder is XYZ." but then in the `decompose` it uses ZYX. Initially I had XYZ and was getting different results. When I found this existing code and switched my new code to ZYX I got the same results and the previous Maya code. I think it has to do with how when we decompose, we must reverse.